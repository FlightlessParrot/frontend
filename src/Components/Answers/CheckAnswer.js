import { Box, Center, Divider, useDisclosure } from "@chakra-ui/react";
import AnswerCategory from "./AnswerCategory";
import HappyCircle from "../Circles/HappyCircle";
import SadCircle from "../Circles/SadCircle";
import NotCountMe from "../Buttons/NotCountMe"
import { useEffect, useState } from "react";
import useShowToast from "../../hooks/useShowToast";


export default function CheckAnswer({question, children, correct=null, generatedTest}) {
    const {isOpen, onToggle }=useDisclosure()
    const [responseOk,setResponseOk]=useState();
    const toast=useShowToast()
    const icon=()=>{ switch(correct)
                {
                    case 1:
                        return <HappyCircle />
                    case 0:
                        return <SadCircle />
                    default: 
                        return null
                }
    }
    useEffect(
    ()=>{     if(responseOk)
        {
            
            toast({title: 'Usunięto ze statystyk.',
        description: `Pytanie "${question.question}" nie będzie liczone na potrzeby statystyk.`,
        statues:'success'})
        setResponseOk(null)
        }
        if(responseOk===false)
        {
            toast({title: 'Nie udało się usunąć ze statystyk',
        description: `Pytanie "${question.question}" wciąż jest liczone na potrzeby statystyk.`,
        status:'error'})
        setResponseOk(null)
        }
    },[responseOk, toast, question.question, setResponseOk]
    )
  return (
 <div className="flex flex-col bg-sel p-4 m-4 max-w-fit rounded" >
    {icon()}
    <AnswerCategory>Pytanie:</AnswerCategory>
    <Divider></Divider>
    <Center>
    <p className="max-w-lg ">{question.question}</p></Center>
    <AnswerCategory pt={[4,8]}>Udzielona odpowiedź</AnswerCategory>
    <Divider/>
    <Box p={[2,4]}>{children}</Box>
    {generatedTest.egzam===0 && question.type==='short-answer' && <NotCountMe testId={generatedTest.id} questionId={question.id} setResponseOk={setResponseOk} />}
    <Divider />
    {question.explanation && (<div>
        <Center><button className="action-button" onClick={onToggle}>{!isOpen ?'Zobacz': 'Schowaj'} wyjaśnienie</button></ Center>
        {isOpen &&
        <div>
        <AnswerCategory pt={[4,8]}>Wyjaśnienie</AnswerCategory>
        <p>{question.explanation}</p>
        
        </div>}

    </div>)}
 </div>
  )
}