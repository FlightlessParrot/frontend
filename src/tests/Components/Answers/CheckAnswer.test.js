import renderer from 'react-test-renderer'
import CheckAnswer from '../../../Components/Answers/CheckAnswer'

describe('test Check Answer Component', ()=>{
    const data={
        question: 'Who is the parrot?',
        answer: 'I hope I am not'
    }
it('test null component render correctly', ()=>{
    const app=renderer.create(<CheckAnswer {...data} ><p>{data.answer}</p></CheckAnswer>).toJSON()
    expect(app).toMatchSnapshot()
})
it('test happy component render correctly', ()=>{
    const app=renderer.create(<CheckAnswer {...data} correct ><p>{data.answer}</p></CheckAnswer>).toJSON()
    expect(app).toMatchSnapshot()
})
it('test sad component render correctly', ()=>{
    const app=renderer.create(<CheckAnswer {...data} correct={false}> <p>{data.answer}</p></CheckAnswer>).toJSON()
    expect(app).toMatchSnapshot()
})

})