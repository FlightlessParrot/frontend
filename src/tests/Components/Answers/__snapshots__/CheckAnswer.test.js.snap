// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Check Answer Component test happy component render correctly 1`] = `
<div
  className="flex flex-col bg-sel p-4   m-4 max-w-fit rounded"
>
  <svg
    className="chakra-icon css-h24txs"
    focusable={false}
    viewBox="0 0 24 24"
  >
    <path
      d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
      fill="currentColor"
    />
  </svg>
  <h2
    className="text-md font-extralight text-zinc-300 text-center m-2"
  >
    Pytanie:
  </h2>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1upb9tn"
  />
  <div
    className="css-gmuwbf"
  >
    <p
      className="max-w-lg "
    >
      Who is the parrot?
    </p>
  </div>
  <h2
    className="text-md font-extralight text-zinc-300 text-center m-2"
  >
    Udzielona odpowiedź
  </h2>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1upb9tn"
  />
  <div
    className="css-k1i3ui"
  >
    <p>
      I hope I am not
    </p>
  </div>
</div>
`;

exports[`test Check Answer Component test null component render correctly 1`] = `
<div
  className="flex flex-col bg-sel p-4   m-4 max-w-fit rounded"
>
  <h2
    className="text-md font-extralight text-zinc-300 text-center m-2"
  >
    Pytanie:
  </h2>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1upb9tn"
  />
  <div
    className="css-gmuwbf"
  >
    <p
      className="max-w-lg "
    >
      Who is the parrot?
    </p>
  </div>
  <h2
    className="text-md font-extralight text-zinc-300 text-center m-2"
  >
    Udzielona odpowiedź
  </h2>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1upb9tn"
  />
  <div
    className="css-k1i3ui"
  >
    <p>
      I hope I am not
    </p>
  </div>
</div>
`;

exports[`test Check Answer Component test sad component render correctly 1`] = `
<div
  className="flex flex-col bg-sel p-4   m-4 max-w-fit rounded"
>
  <div
    className="rounded-full p-4 bg-red-600 w-[60px] h-[60px] flex justify-center items-center"
  >
    <svg
      className="chakra-icon css-dhtqli"
      focusable={false}
      viewBox="0 0 24 24"
    >
      <path
        d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
        fill="currentColor"
      />
    </svg>
  </div>
  <h2
    className="text-md font-extralight text-zinc-300 text-center m-2"
  >
    Pytanie:
  </h2>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1upb9tn"
  />
  <div
    className="css-gmuwbf"
  >
    <p
      className="max-w-lg "
    >
      Who is the parrot?
    </p>
  </div>
  <h2
    className="text-md font-extralight text-zinc-300 text-center m-2"
  >
    Udzielona odpowiedź
  </h2>
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1upb9tn"
  />
  <div
    className="css-k1i3ui"
  >
     
    <p>
      I hope I am not
    </p>
  </div>
</div>
`;
